packages/peregrine/lib/talons/AddressBookPage/useAddressBookPage.js

```diff
@@ -70,7 +70,9 @@ export const useAddressBookPage = (props = {}) => {
     const [isDialogOpen, setIsDialogOpen] = useState(false);
     const [activeEditAddress, setActiveEditAddress] = useState();
     const isDialogEditMode = !!activeEditAddress;
-    const [formApi, setFormApi] = useState();
+    const [formInitialValues, setFormInitialValues] = useState({
+        country_code: 'US'
+    });
```

Comment:
You may not need a duplicate piece of state here; pass `activeEditAddress` as initialValues and update `handleAddAddress` to pass this default value when it calls `setActiveEditAddress`, eg. `setActiveEditAddress({ country_code: 'US' });`.

Regardless of this duplication, I think there is a bug here where `initialValues` is not being set correctly on add, and the previous edited address is being displayed. Either go with my proposal above, or add `setFormInitialValues({ country_code: 'US' });` to that handler if there's a necessity to have two copies of this state.

Reply:
Dang, you're totally right - great catch